dt-metrics/combined/date-range-activity.php

```diff
@@ -0,0 +1,370 @@
+                    'total_label' => __( 'Total', 'disciple_tools' ),
+                    'date_select_label' => __( 'Date Range', 'disciple_tools' ),
+                    'submit_button_label' => __( 'Reload', 'disciple_tools' ),
+                    'results_table_head_title_label' => __( 'Title', 'disciple_tools' ),
+                    'results_table_head_date_label' => __( 'Date', 'disciple_tools' ),
+                    'results_table_head_new_value_label' => __( 'New Value', 'disciple_tools' ),
+                    'regions_of_focus' => __( 'Regions of Focus', 'disciple_tools' ),
+                    'all_locations' => __( 'All Locations', 'disciple_tools' )
+                ],
+                'select_options' => [
+                    'post_type_select_options' => $this->post_type_select_options,
+                    'post_field_select_options' => $this->post_field_select_options,
+                ],
+                'field_settings' => $this->field_settings
+            ]
+        );
+    }
+
+    public function add_api_routes() {
+        $version   = '1';
+        $namespace = 'dt/v' . $version;
+
+        register_rest_route(
+            $namespace, '/metrics/field_settings/(?P<post_type>\w+)', [
+                [
+                    'methods'  => WP_REST_Server::READABLE,
+                    'callback' => [ $this, 'field_settings' ],
+                    'permission_callback' => '__return_true',
+                ],
+            ]
+        );
+
+        register_rest_route(
+            $namespace, '/metrics/render_field_html', [
+                [
+                    'methods'  => WP_REST_Server::READABLE,
+                    'callback' => [ $this, 'render_field_html' ],
+                    'permission_callback' => '__return_true'
+                ]
+            ]
+        );
+
+        register_rest_route(
+            $namespace, '/metrics/dummy_endpoint', [
+                [
+                    'methods'  => WP_REST_Server::READABLE,
+                    'callback' => [ $this, 'dummy_endpoint' ],
+                    'permission_callback' => '__return_true',
+                ],
+            ]
+        );
+
+        register_rest_route(
+            $namespace, '/metrics/date_range_activity', [
+                [
+                    'methods'  => WP_REST_Server::CREATABLE,
+                    'callback' => [ $this, 'date_range_activity' ],
+                    'permission_callback' => '__return_true',
+                ],
+            ]
+        );
+    }
+
+    public function field_settings( WP_REST_Request $request ) {
+        if ( !$this->has_permission() ) {
+            wp_send_json_error( new WP_Error( 'get_field_settings', 'Missing Permissions', [ 'status' => 400 ] ) );
+        }
+        $url_params = $request->get_url_params();
+        return $this->get_field_settings( $url_params['post_type'] );
+    }
+
+    public function render_field_html( WP_REST_Request $request ){
```

Comment:
@kodinkat how can we steer away from sending html in API calls? 
It's a security risk as the result can't be escaped

Reply:
Could capture all field html when initially loading the page; or, refresh page and load on refresh... Personally, I'm not keen on either approach... Alternatively, html could be generated within frontend, but this too would add to code / logic duplication.